test/gomega/matchers/have_protocol.go

```diff
@@ -0,0 +1,34 @@
+package matchers
+
+import (
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gstruct"
+	"github.com/onsi/gomega/types"
+	"github.com/solo-io/gloo/test/gomega/transforms"
+)
+
+const (
+	HTTP1Protocol = "HTTP/1.1"
+	HTTP2Protocol = "HTTP/2"
+)
+
+// HaveProtocol expects an http response with the given protocol
+func HaveProtocol(protocol string) types.GomegaMatcher {
+	if protocol == "" {
+		// If protocol is defined, we create a matcher that always succeeds
```

Comment:
nit: If protocol is _not_ defined

Plus, there is a reference in the comment about "an And matcher for 0 objects". To me that feels like an implementation detail of how this matcher is consumed, and it might mislead people

Reply:
Thanks, I've updated this to be more clear